modules/tracing/mule-tracer-internal-impl/src/main/java/org/mule/runtime/tracer/impl/context/NoExportOnEndInternalSpan.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.tracer.impl.context;
+
+import static org.mule.runtime.tracer.api.span.ExportableInternalSpan.asExportable;
+import static org.mule.runtime.tracer.impl.clock.Clock.getDefault;
+
+import org.mule.runtime.api.profiling.tracing.Span;
+import org.mule.runtime.api.profiling.tracing.SpanDuration;
+import org.mule.runtime.api.profiling.tracing.SpanError;
+import org.mule.runtime.api.profiling.tracing.SpanIdentifier;
+import org.mule.runtime.tracer.api.span.ExportableInternalSpan;
+import org.mule.runtime.tracer.api.span.InternalSpan;
+import org.mule.runtime.tracer.api.span.error.InternalSpanError;
+import org.mule.runtime.tracer.api.span.exporter.SpanExporter;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+/**
+ * A {@link InternalSpan} wrapper to prevent exporting on end.
+ *
+ * @since 4.5.1, 4.6.0
+ */
+public class NoExportOnEndInternalSpan implements ExportableInternalSpan {
```

Comment:
We are not using inheritance but the ExecutionSpan is practically the same than this one, but also exporting at the end() call, right? Can we name them more closely?

Reply:
I've renamed ExecutionSpan to ExportOnEndExecutionSpan